src/sysfiles.c

```diff
@@ -677,8 +680,11 @@ Int SyFopen (
 #endif
 
     /* try to open the file                                                */
-    syBuf[fid].fp = open(name,flags, 0644);
-    if ( 0 <= syBuf[fid].fp ) {
+    if (endsgz && (syBuf[fid].gzfp = gzopen(name, mode))) {
+        syBuf[fid].type = gzip_socket;
+        syBuf[fid].fp = -1;
+        syBuf[fid].bufno = -1;
+    } else if (0 <= (syBuf[fid].fp = open(name, flags, 0644))) {
```

Comment:
Hm, so we are patching the most low-level function in the kernel for opening a file to transparently access .gz files. That makes it really tough to bypass this, one has to either bypass `SyFopen` or else copy or rename the file in question to look at its "true" content.

I think it is OK to do this now in this PR, but I'd like us to revisit this. I notice that `SyFopen` still takes an old clunky "mode" string as second argument. That argument could be augmented to indicate that "raw" or "binary" mode (no automatic (de)compressing) is wanted. Indeed, we already supported on Cygwin using `"rb"` and `"wb"` modes (for savefiles).

But I'd also suggest to change it from a `const char *` to an integer bit mask (basically similar to the POSIX `O_RDONLY` etc. flags for `open`, just out own portable version of those).

Anyway, either way it'd be trivial to add a simple kernel helper function which basically does the same as `StringFile` but using this "raw" mode (`StringFile` is a GAP function but one that's trivial to rewrite in a <10 lines kernel function, and then only the argument to `SyFopen` in there needs to be changed)

Reply:
Of course the idea of changing the mode string to an int (or enum) is separate from the rest; this is just ramblings, don't feel obliged to implement any of this.

Reply:
I'm not opposed to how to fix this, but I'm a bit unsure about implementing it at all. I don't really like adding stuff just to support testing. If anyone is upset they can't access raw gzip files I'm very happy to fix it -- although at the moment we don't really support binary file access at all (on Windows we always do line ending conversion for example, so we'd probably corrupt raw gzipped files anyway).